openrr-planner/src/funcs.rs

```diff
@@ -148,15 +148,22 @@ pub fn interpolate<T>(
 where
     T: Float,
 {
-    let mut times = Vec::new();
     let key_frame_unit_duration = total_duration / (T::from(points.len())? - T::one());
+    let mut times = Vec::with_capacity(points.len());
     for i in 0..points.len() {
         times.push(key_frame_unit_duration * T::from(i)?);
     }
+    /* FIXME:
+     * 	let time = (0_usize..points.len())
+     *					.map( |i| T::from(i)*key_frame_unit_duration)
+     *					.collect::<Vec<f64>>();
+     * Another purpose
+     * I don't compare and I've not know which is more efficiently yet.
+     */
```

Comment:
I think you can use your method! At least it reduces `mut`.

Reply:
I expect that the conventional method needs `mut` .
My method is not changed later process so it is not necessary  `mut`.
I think.

---------------------------
I'll update it. Wait a minute, please.

Reply:
I expect that the conventional method needs `mut` .
My method is not changed later process so it is not necessary  `mut`.
I think.

---------------------------
I'll update it. Wait a minute, please.